app/controllers/recipe_ingredients_controller.rb

```diff
@@ -0,0 +1,9 @@
+class RecipeIngredientsController < ApplicationController
+  def index
+    @recipe = Recipe.find(params[:recipe_id])
+    @ingredient = Ingredient.find(params[:ingredient_id])
+    @ingredients = @recipe.ingredients
+  end
+
+  
+end
```

Comment:
I'm a little confused for what this controller is trying to do. It looks like you want to grab all of a recipes ingredients and display that on the page, but we really don't want to do that in this controller. Especially since we can just grab a recipe's ingredients from the association a recipe has with ingredients similar to how you've done it on line 5.

In the `RecipesController` `show` action, once we find the recipe, through our association on the `Recipe` model to ingredients, we can make use of that association to return all the ingredients for that recipe.

```ruby
class RecipesController < ApplicationController
  def show
    @recipe = Recipe.find(params[:id])
  end
end
```

```erb
<% @recipe.ingredients.each do |ingredient| %>
```

☝️ Notice how we are making use of the association between a recipe and it's ingredients.

---

This controller is not necessary.